### PR TITLE
Fix the bug when extracting error detail

### PR DESCRIPTION
**Pull request by sachinira**
# Description

$subject

# Checklist:

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: sfdc/errors.bal [hunk 1]
```diff
@@ -23,16 +23,21 @@ public type Error error<ErrorDetails>;
...
-const string HTTP_CLIENT_ERROR = "Failed to establish the communication with the upstream server or a data binding failure. Refer error.cause()";
+const string HTTP_CLIENT_ERROR = "Failed to establish the communication with the upstream server or a data binding failure. Refer error.cause() for more details";
 public const string ERR_EXTRACTING_ERROR_MSG = "Error occured while extracting errors from payload.";
 public const string HTTP_ERROR_MSG = "Error occurred while getting the HTTP response.";
 public const string INVALID_CLIENT_CONFIG = "Invalid values provided for client configuration parameters.";
+const STATUS_CODE = "statusCode";
+const HEADERS = "headers";
+const BODY = "body";
+
+
```
Comment (LakshanSS) [inline sfdc/errors.bal:43]: Shall we remove these unnecessary new lines?

File: sfdc/rest_utils.bal [hunk 2]
```diff
@@ -16,6 +16,7 @@
 import ballerina/log;
 import ballerina/http;
 import ballerina/url;
+// import ballerina/io;
```
Comment (LakshanSS) [inline sfdc/rest_utils.bal:19]: Shall we remove this commented line?

File: sfdc/tests/rest_api_test.bal [hunk 3]
```diff
@@ -51,11 +52,12 @@ string testRecordId = "";
 function testCreateRecord() {
     log:printInfo("baseClient -> createRecord()");
     string|Error stringResponse = baseClient->createRecord(ACCOUNT, accountRecord);
-
     if (stringResponse is string) {
         test:assertNotEquals(stringResponse, "", msg = "Found empty response!");
         testRecordId = stringResponse;
     } else {
+        io:print(stringResponse.cause());
+
```
Comment (LakshanSS) [inline sfdc/tests/rest_api_test.bal:60]: Shall we remove this newline?

**Review by LakshanSS — APPROVED** [review]
Approved with minor comments

File: sfdc/errors.bal (hunk 1 above)
Comment (sachinira) [reply sfdc/errors.bal]: Removed. thanks

File: sfdc/rest_utils.bal (hunk 2 above)
Comment (sachinira) [reply sfdc/rest_utils.bal]: Removed.

File: sfdc/tests/rest_api_test.bal (hunk 3 above)
Comment (sachinira) [reply sfdc/tests/rest_api_test.bal]: Removed. thanks

**Pull request merged**